### ISSUE TITLE
Include error messages from dry run json report in stdout when knapsack_pro:rspec_test_example_detector fails

### ISSUE THREAD

**Issue by cccCody** [KnapsackPro/knapsack_pro-ruby#170]
I'm trying to enable `KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES`, but every node in CI fails with the [error here](https://github.com/KnapsackPro/knapsack_pro-ruby/blob/1ba4f4bf6aa7836391d477032ed256edc540a034/lib/knapsack_pro/base_allocator_builder.rb#L52-L53). The "actionable error message" above the error suggests running a command on the CI machine to reproduce the problem, e.g.
```
To reproduce the error triggered by the RSpec, please try to run below command (this way,
you can find out what is causing the error):
bundle exec rspec --format json --dry-run --out \
.knapsack_pro/test_case_detectors/rspec/rspec_dry_run_json_report_node_0.json \
--default-path spec foo.rb bar.rb
```
When I do, the command completes successfully, every time, so I can't tell why it's failing during automated runs. It's really frustrating to not be able to reproduce the issue. Why not capture the output of the command in knapsack and include it with the error message if that command fails?

**Comment by ArturT** [comment]
What's the content of the `.knapsack_pro/test_case_detectors/rspec/rspec_dry_run_json_report_node_0.json` file?
Was it generated on the disk when you run the command in the CI environment?

Was the `.knapsack_pro` directory generated on the CI? Maybe it's a permissions error.

What CI provider do you use?


> Why not capture the output of the command in knapsack and include it with the error message if that command fails?

This is something to investigate. I was wondering about printing the error a long time ago, but it was more complex.
There were a few approaches to this and how to make it secure. https://stackoverflow.com/questions/690151/getting-output-of-system-calls-in-ruby 

The `bundle exec rspec --format json --dry-run --out ...` command writes a file on the disk, and most of the time, it does not produce any output, so there was no reason to capture the output.


Please share the full output of the error that you see. You can also email our support at https://knapsackpro.com/contact if the error contains private data (not to share it publicly here).

**Comment by cccCody** [comment]
The files do get generated when I run the command manually. I will send a few samples of them to you by e-mail along with the build log since they might be considered sensitive.

We're using the latest version of Jenkins LTS.

**Comment by cccCody** [comment]
I took a look at a couple of nodes and saw the file being generated, even though it was reported as having failed.
I also see that they look like they have the same content on different nodes, even though I'd expect them to list different tests. For example, on the same run, both `slow_test_files_node_21.json` and `slow_test_files_node_26.json` have the same content:
```
[{"path":"spec/features/spaces/account_creation/onboarding_visit_public_network_course_spec.rb","time_execution":2227.537136728004},{"path":"spec/features/onboarding_visit_public_network_course_spec.rb","time_execution":2157.940042964002}]
```
![Screen Shot 2022-06-21 at 3 13 29 PM](https://user-images.githubusercontent.com/753031/174906256-efc0283b-301a-43dd-b2df-1916ff3fdc6b.png)

**Comment by cccCody** [comment]
Aha! Saving the artifact gets me this:
```
{
	"version": "3.8.2",
	"messages": ["\nAn error occurred while loading ./spec/features/spaces/account_creation/onboarding_visit_public_network_course_spec.rb.\nFailure/Error: require_dependency('premailer') # premailer must be manually invoked for batch renders. premailer inlines the email css\n\nLoadError:\n  No such file to load -- premailer\n# ./app/mailers/batch_mailer.rb:2:in `\u003ctop (required)\u003e'\n# ./app/mailers/batch_invite_mailer.rb:1:in `\u003ctop (required)\u003e'\n# ./config/environment.rb:9:in `\u003ctop (required)\u003e'\n# ./spec/spec_helper.rb:2:in `\u003ctop (required)\u003e'\n# ./spec/features/spaces/account_creation/onboarding_visit_public_network_course_spec.rb:1:in `\u003ctop (required)\u003e'\n", "\nAn error occurred while loading ./spec/features/onboarding_visit_public_network_course_spec.rb.\nFailure/Error: require File.expand_path('../../config/environment', __FILE__)\n\nRuntimeError:\n  can't modify frozen Array\n# ./config/environment.rb:9:in `\u003ctop (required)\u003e'\n# ./spec/spec_helper.rb:2:in `\u003ctop (required)\u003e'\n# ./spec/features/onboarding_visit_public_network_course_spec.rb:1:in `\u003ctop (required)\u003e'\n", "No examples found."],
	"examples": [],
	"summary": {
		"duration": 4.3535e-05,
		"example_count": 0,
		"failure_count": 0,
		"pending_count": 0,
		"errors_outside_of_examples_count": 2
	},
	"summary_line": "0 examples, 0 failures, 2 errors occurred outside of examples"
}
```

**Comment by cccCody** [comment]
I still think it would be helpful to print those messages out along with the top level error message, rather than encouraging users to repro and then go digging through internal knapsack files.

**Comment by ArturT** [comment]
I've prepared the improvement to show the content of the JSON report so that it's easier to debug an error when the error message is printed only inside of the JSON report.
https://github.com/KnapsackPro/knapsack_pro-ruby/pull/172

**Comment by ArturT** [comment]
@cccCody I've released the knapsack_pro gem 3.3.0 version.
